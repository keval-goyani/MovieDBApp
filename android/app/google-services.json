{
  "project_info": {
    "project_number": "89996564819",
    "project_id": "moviedb-2908d",
    "storage_bucket": "moviedb-2908d.appspot.com"
  },
  "client": [
    {
      "client_info": {
        "mobilesdk_app_id": "1:89996564819:android:e942630a87714b1d3ea9a6",
        "android_client_info": {
          "package_name": "com.moviedbapp"
        }
      },
      "oauth_client": [
        {
          "client_id": "89996564819-hph4f8e5b765gev43v87vj04t9tun4il.apps.googleusercontent.com",
          "client_type": 3
        }
      ],
      "api_key": [
        {
          "current_key": "AIzaSyBYrM1MC9Go35l8ra748zqBtCRDkrB_Nn4"
        }
      ],
      "services": {
        "appinvite_service": {
          "other_platform_oauth_client": [
            {
              "client_id": "89996564819-hph4f8e5b765gev43v87vj04t9tun4il.apps.googleusercontent.com",
              "client_type": 3
            }
          ]
        }
      }
    }
  ],
  "configuration_version": "1"
}