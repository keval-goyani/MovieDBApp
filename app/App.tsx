import React, {useEffect} from 'react';
import {SafeAreaView, Text} from 'react-native';
import SplashScreen from 'react-native-splash-screen';
import {styles} from './theme';

const App = () => {
  useEffect(() => {
    SplashScreen.hide();
  }, []);

  return (
    <SafeAreaView style={styles.container}>
      <Text>Hello World</Text>
    </SafeAreaView>
  );
};

export default App;
