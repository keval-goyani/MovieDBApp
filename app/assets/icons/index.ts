export default {
  menuIcon: require('./menuIcon.png'),
  searchIcon: require('./searchIcon.png'),
  movieDbIcon: require('./movieDBIcon.png'),
  expandIcon: require('./expandIcon.png'),
  threeDotIcon: require('./threeDotIcon.png'),
  playIcon: require('./playIcon.png'),
  backIcon: require('./backIcon.png'),
  movieDbIcon3x: require('./movieDB3x.png'),
  movieDbIcon2x: require('./movieDB2x.png'),
  homeIcon: require('./homeIcon.png'),
  trailerIcon: require('./trailerIcon.png'),
  moviesIcon: require('./moviesIcon.png'),
  communityIcon: require('./communityIcon.png'),
  notFound: require('./notFoundIcon.png'),
  comingSoon: require('./comingSoon.png'),
  logout: require('./logout.png'),
  trending: require('./trending.png'),
  tv: require('./tv.png'),
  avatar: require('./user.png'),
  chatBackground: require('./chatBackground.png'),
  send: require('./send.png'),
  location: require('./share.png'),
  camera: require('./camera.png'),
  attach: require('./attach.png'),
  gallery: require('./gallery.png'),
  dotMenu: require('./dots.png'),
};
