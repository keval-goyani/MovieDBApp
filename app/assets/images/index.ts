export default {
  docFirstPage: require('./docFirstPage.png'),
};
