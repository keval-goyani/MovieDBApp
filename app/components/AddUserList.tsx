import { useFocusEffect, useNavigation } from '@react-navigation/native';
import React, { useCallback, useEffect, useState } from 'react';
import { FlatList, Text, TouchableOpacity, View } from 'react-native';
import Check from 'react-native-vector-icons/AntDesign';
import { useDispatch, useSelector } from 'react-redux';
import { asMutable } from 'seamless-immutable';
import { ProfileImage, UserListEmpty } from '../components';
import {
  AddUserListProps,
  appConstants,
  NavigationDataType,
  navigationStrings,
  RenderItemTypes,
  SelectedUsersProps,
  strings,
  UserDataType,
  UsersDocumentDataType,
} from '../constants';
import { authDataSelectors } from '../redux/AuthRedux';
import userListDataAction, { userListSelector } from '../redux/UserListRedux';
import { conversationIdCreation } from '../services';
import { Color, moderateScale } from '../theme';
import { styles } from './styles/AddUserListStyles';

const AddUsersList = ({ userListData, setSelectedUsers }: AddUserListProps) => {
  const navigation: NavigationDataType = useNavigation();
  const dispatch = useDispatch();
  const { user } = useSelector(authDataSelectors.getData);
  const { fetchingUserList } = useSelector(userListSelector.getData);
  const [selectedItems, setSelectedItems] = useState<UserDataType[]>([]);

  const fetchingUser = useCallback(() => {
    appConstants.userRef.onSnapshot(users => {
      const fireStoreUserList: UsersDocumentDataType[] = [];

      users.forEach(userData => {
        if (userData?.data()?.uid !== user?.uid) {
          fireStoreUserList.push(userData?.data());
        }

        dispatch(userListDataAction.usersListSuccess(fireStoreUserList));
      });
    });
  }, [dispatch, user]);

  useFocusEffect(
    useCallback(() => {
      const sub = fetchingUser();
      return () => sub;
    }, [fetchingUser]),
  );

  useEffect(() => {
    setSelectedUsers(selectedItems);
  }, [selectedItems, setSelectedUsers]);

  const renderUserList = ({ item }: RenderItemTypes) => {
    const usersEmail = [user?.email ?? '', item?.email];
    const conversationId: string = conversationIdCreation(usersEmail);

    const handleOnPress = (selectedUser: UserDataType) => {
      if (selectedItems.length) {
        return selectUsers(selectedUser);
      }
      navigation.goBack();
      navigation.navigate(navigationStrings.Chat, {
        conversationId,
        username: item?.username,
        receiverId: item?.uid,
        userStatus: item?.status,
        profileImage: item?.profileImage,
        userEmail: item?.email,
      });
    };

    const getSelected = (usersData: SelectedUsersProps) =>
      selectedItems.includes(usersData);

    const selected = getSelected(item);

    const selectUsers = (selectedUser: UserDataType) => {
      if (selectedItems.includes(selectedUser)) {
        const filteredList = selectedItems.filter(
          (items: UserDataType) => items !== selectedUser,
        );
        return setSelectedItems(filteredList);
      }
      setSelectedItems([...selectedItems, selectedUser]);
    };

    return (
      <TouchableOpacity
        style={styles.listItem}
        onPress={() => handleOnPress(item)}
        activeOpacity={0.8}
        onLongPress={() => selectUsers(item)}>
        <View style={styles.itemContainer}>
          <ProfileImage
            profileImage={item?.profileImage}
            userStatus={item?.status}
          />
          <View style={styles.nameView}>
            <Text style={styles.text}>{item?.username}</Text>
            <Text
              style={styles.lastChatText}
              ellipsizeMode="tail"
              numberOfLines={1}>
              {item?.email}
            </Text>
          </View>
        </View>
        {selected && (
          <>
            <View style={styles.overlay} />
            <Check
              name={strings.checkIcon}
              size={moderateScale(23)}
              color={Color.lightBlue}
              style={styles.iconStyle}
            />
          </>
        )}
      </TouchableOpacity>
    );
  };

  return (
    <FlatList
      data={asMutable(userListData) ?? []}
      renderItem={renderUserList}
      ListEmptyComponent={
        <UserListEmpty
          fetching={fetchingUserList}
          userListLength={userListData.length}
        />
      }
      showsVerticalScrollIndicator={false}
      bounces={false}
    />
  );
};

export default AddUsersList;
