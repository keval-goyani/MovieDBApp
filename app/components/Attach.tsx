import { useNavigation } from '@react-navigation/native';
import React, { FC } from 'react';
import { Alert, View } from 'react-native';
import SystemSetting from 'react-native-system-setting';
import { CustomIconRounder } from '../components';
import {
  AttachDataType,
  NavigationDataType,
  navigationStrings,
  strings,
} from '../constants';
import {
  handleCameraPermission,
  handleDocumentPermission,
  handleGalleryPermission,
} from '../services';
import { Icons } from '../theme';
import styles from './styles/AttachStyles';

const Attach: FC<AttachDataType> = ({
  setIsAttach,
  setImagePath,
  setDocumentData,
  conversationId,
  username,
  receiverId,
}) => {
  const navigation: NavigationDataType = useNavigation();

  const turnOnLocation = (enable: boolean) => {
    if (enable) {
      navigation.navigate(navigationStrings.Location, {
        isFromChat: false,
        ...{ conversationId, username, receiverId },
      });
      setIsAttach(false);
    } else {
      Alert.alert(strings.turnOn, strings.goToSetting, [
        {
          onPress: () => SystemSetting.switchLocation(),
          text: strings.setting,
        },
        { text: strings.cancel },
      ]);
    }
  };

  const locationHandler = () => {
    SystemSetting.isLocationEnabled()
      .then(enable => {
        turnOnLocation(enable);
      })
      .catch(error => Alert.alert(strings.warning, error));
  };

  return (
    <View style={styles.container}>
      <View style={styles.iconContainer}>
        <CustomIconRounder
          backgroundColor={styles.documentBackgroundColor}
          path={Icons.fileIcon}
          iconName={strings.document}
          tintColor={styles.documentTintColor}
          onPress={() => {
            setIsAttach(false);
            handleDocumentPermission(setDocumentData);
          }}
        />
        <CustomIconRounder
          backgroundColor={styles.cameraBackgroundColor}
          path={Icons.cameraIcon}
          iconName={strings.camera}
          tintColor={styles.cameraTintColor}
          onPress={() => {
            handleCameraPermission(setImagePath);
            setIsAttach(false);
          }}
        />
        <CustomIconRounder
          backgroundColor={styles.galleryBackgroundColor}
          path={Icons.galleryIcon}
          iconName={strings.gallery}
          tintColor={styles.galleryTintColor}
          onPress={() => {
            handleGalleryPermission(setImagePath);
            setIsAttach(false);
          }}
        />
        <CustomIconRounder
          backgroundColor={styles.locationBackgroundColor}
          path={Icons.locationIcon}
          iconName={strings.map}
          tintColor={styles.locationTintColor}
          onPress={locationHandler}
        />
      </View>
    </View>
  );
};

export default Attach;
