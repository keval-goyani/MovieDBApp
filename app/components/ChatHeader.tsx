import { useNavigation } from '@react-navigation/native';
import React from 'react';
import { Image, Pressable, Text, TouchableOpacity, View } from 'react-native';
import { ChatMenu } from '../components';
import { ChatHeaderDataType, NavigationDataType } from '../constants';
import { Icons } from '../theme';
import { styles } from './styles/ChatHeaderStyles';

const ChatHeader = ({
  username,
  picture,
  onlineStatus,
  showMenu,
  setShowMenu,
  setIsAttach,
  setCameraModal,
  setChatWallpaper,
  receiverId,
  conversationId,
}: ChatHeaderDataType) => {
  const navigation: NavigationDataType = useNavigation();

  return (
    <>
      <View style={styles.container}>
        <TouchableOpacity onPress={() => navigation.goBack()}>
          <Image source={Icons.backIcon} style={styles.leftIconStyle} />
        </TouchableOpacity>
        <View style={styles.profileView}>
          <View style={styles.profile}>
            <Image source={picture} style={styles.avatarImage} />
            <View style={styles.userNameAndOnlineStatus}>
              <Text style={styles.username}>{username}</Text>
              <Text style={styles.onlineStatus}>{onlineStatus}</Text>
            </View>
          </View>
        </View>
        <Pressable
          onPress={() => {
            setShowMenu(!showMenu);
            setIsAttach(false);
            setCameraModal(false);
          }}>
          <Image source={Icons.dotMenu} style={styles.dotsMenu} />
        </Pressable>
      </View>
      {showMenu && (
        <ChatMenu
          {...{ conversationId, receiverId, setChatWallpaper, setShowMenu }}
        />
      )}
    </>
  );
};

export default ChatHeader;
