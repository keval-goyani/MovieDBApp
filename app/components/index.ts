import ChatHeader from './ChatHeader';
import ChatInput from './ChatInput';
import CustomSkeleton from './CustomSkeleton';
import DropDownMenu from './DropDownMenu';
import Header from './Header';
import List from './List';
import ListContainer, { movieListItem } from './ListContainer';
import Loader from './Loader';
import LoadingState from './LoadingState';
import Message from './Message';
import MessageList from './MessageList';
import Movies from './Movies';
import MovieTrailer, { trailerListItem } from './MovieTrailers';
import SearchModal from './SearchModal';
import SkeletonCard from './SkeletonCard';
import TabBarIcon from './TabBarIcon';
import Trailers from './Trailers';
import UserListEmpty from './UserListEmpty';
import UsersList from './UsersList';

export {
  ChatHeader,
  ChatInput,
  DropDownMenu,
  Header,
  List,
  ListContainer,
  MovieTrailer,
  Message,
  MessageList,
  TabBarIcon,
  SearchModal,
  Loader,
  LoadingState,
  Movies,
  movieListItem,
  Trailers,
  trailerListItem,
  UsersList,
  SkeletonCard,
  CustomSkeleton,
  UserListEmpty,
};
