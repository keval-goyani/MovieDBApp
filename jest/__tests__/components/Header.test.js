import { fireEvent, render } from '@testing-library/react-native';
import React from 'react';
import { Icons } from '../../../app/assets';
import { Header } from '../../../app/components';

describe('Header', () => {
  it('header component snapshot', () => {
    const { toJSON } = render(<Header />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('should be pressed plusButton', () => {
    const { getByTestId } = render(
      <Header
        title="header"
        rightIcon={Icons.movieDbIcon}
        isFromChatMessageScreen={true}
      />,
    );
    const button = getByTestId('button');
    fireEvent.press(button);
  });

  it('should be open search modal', () => {
    const { getByTestId } = render(
      <Header rightIcon={Icons.movieDbIcon} isFromChatMessageScreen={false} />,
    );
    const button = getByTestId('button');
    fireEvent.press(button);
  });
});
