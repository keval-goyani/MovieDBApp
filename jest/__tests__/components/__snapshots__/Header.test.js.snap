// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header header component snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": undefined,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "paddingHorizontal": undefined,
      "paddingVertical": undefined,
    }
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Image
      style={
        Object {
          "height": undefined,
          "tintColor": undefined,
          "width": undefined,
        }
      }
    />
  </View>
  <Image />
  <View
    style={
      Object {
        "height": undefined,
        "tintColor": undefined,
        "width": undefined,
      }
    }
  />
</View>
`;
